shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -1727,7 +1710,8 @@ - (CGRect)caretRectForPosition:(UITextPosition*)position {
   // For both cases, return a zero-width rectangle along the downstream edge of the character
   // before the caret position.
   CGRect characterBeforeCaret = rects[0].rect;
-  if ([self isRTLAtPosition:index - 1]) {
+  if ([rects[0] isKindOfClass:[FlutterTextSelectionRect class]] &&
```

Comment:
I think we can assert the class type here?

Reply:
Not sure what you mean. I am checking the class type?

Reply:
Something like: 
```
FlutterTextSelectionRect *rect = (FlutterTextSelectionRect *)rects[0];
if (rect...)
```
Because the rect must be the correct type right? There is no way `isKindOfClass` returns false, am I right?

Reply:
Technically it's possible, this method signature is not `FlutterTextSelectionRect*`, but `UITextSelectionRect*`.